src/installer.ts

```diff
@@ -87,6 +87,7 @@ export async function getJava(
   }
 
   let extendedJavaHome = 'JAVA_HOME_' + version + '_' + arch;
+  extendedJavaHome = extendedJavaHome.toUpperCase().replace(/[^0-9A-Z_]/g, '_');
```

Comment:
My only suggestion would be to add a comment here as to why this is being done. Better to have some comment/link in the code vs hunting down a PR later

Reply:
Makes sense. Done!